src/front/spv/mod.rs

```diff
@@ -2608,7 +2631,34 @@ impl<I: Iterator<Item = u32>> Parser<I> {
                         ctx.bodies.push(body);
                         ctx.body_for_label.entry(target).or_insert(case_body_idx);
 
-                        cases.push((literal as i32, case_body_idx));
+                        // Register this target block id as already having been processed and
+                        // the respective body index assigned and the first case value
+                        self.switch_cases
+                            .insert(target, (case_body_idx, vec![literal as i32]));
+                    }
+
+                    // Loop trough the collected target blocks creating a new case for each
+                    // literal pointing to it, only one case will have the true body and all the
+                    // others will be empty falltrough so that they all execute the same body
+                    // without duplicating code.
+                    //
+                    // Since `switch_cases` is an indexmap the order of insertation is preserved
+                    // this is needed because spir-v defines falltrough order in the switch
+                    // instruction.
+                    let mut cases = Vec::new();
+                    for &(case_body_idx, ref literals) in self.switch_cases.values() {
+                        let value = literals[0];
+
+                        for literal in literals.iter().skip(1).copied() {
```

Comment:
nit: the `.copied()` here is strange. Wouldn't it be easier to just do `for &literal in ...` instead?